packages/kas/src/nodes.ts

```diff
@@ -2931,18 +2933,17 @@ export class Func extends Sym {
         return this.symbol + "(" + this.arg.tex() + ")";
     }
 
-    eval(vars: Vars = {}, options?: ParseOptions) {
+    eval(vars: Vars = {}, options?: ParseOptions): number {
         var arg = this.arg;
         var func = vars[this.symbol];
         var newVars = _.extend(_.clone(vars), {
             x: arg.eval(vars, options),
         });
         var parsedFunc = parse(func, options);
-        if (parsedFunc.parsed) {
-            return parsedFunc.expr.eval(newVars, options);
+        if (!parsedFunc.parsed) {
+            throw new Error(parsedFunc.error);
         }
-        // If parsedFunc isn't actually parsed, return its error
-        return parsedFunc;
```

Comment:
The expectation is that `eval()` always returns a number so I made this function throw instead.  We should probably have a variant of this that returns the `parsedFunc` object.  Then callers in other modules can call `safeEval()`.  We can restrict the use of unsafe `eval()` by making the method private.

TODO: Since that seems like a lot of work, I think for now, the best course of action would be to revert these changes, add a `@ts-expect-error`, and file a ticket.